### DISCUSSION TITLE
Github secrets fails in single workflow

### DISCUSSION THREAD

**Discussion by bfloch** [community/community#25601] [Actions]
<p>This action step involving a secret works in one workflow but fails in another:</p>
<pre><code>${gh_user} = ("${{ github.repository }}" -Split '/')[0]
docker login -u ${gh_user} -p ${{ secrets.DOCKERHUB_TOKEN }}
</code></pre>
<p>Results in:</p>
<pre><code>Run ${gh_user} = ("nerdvegas/rez" -Split '/')[0]
flag needs an argument: 'p' in -p
See 'docker login --help'.
##[error]Process completed with exit code 1.
</code></pre>
<p>The very same workflows worked fine on my local fork of the repo.</p>
<p>This works:</p>
<p><a href="https://github.com/nerdvegas/rez/blob/master/.github/workflows/windows-docker-image.yaml#L42" rel="nofollow noopener">https://github.com/nerdvegas/rez/blob/master/.github/workflows/windows-docker-image.yaml#L42</a></p>
<p>This does not:</p>
<p><a href="https://github.com/nerdvegas/rez/blob/master/.github/workflows/windows-docker.yaml#L43" rel="nofollow noopener">https://github.com/nerdvegas/rez/blob/master/.github/workflows/windows-docker.yaml#L43</a></p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>I checked your two workflow yaml files, and workflow run logs.<br>
There is the step snippet you used in windows-docker.yaml:</p>
<pre><code>- name: Login to docker repository
      run: |
        ${gh_user} = ("${{ github.repository }}" -Split '/')[0]
        '${{ secrets.DOCKERHUB_TOKEN }}' | docker login -u ${gh_user} --password-stdin
</code></pre>
<p>I found <strong>${{ secrets.DOCKERHUB_TOKEN }}</strong> was empty which caused Login to docker reposity step fail.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/0ba07351248dab72f7ebb288de0f0aa7c831cf07.png" data-download-href="/uploads/short-url/1ER0Ykb3RbDnMiPrWE9voy7pKqr.png?dl=1" title="secret logs.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181091963-3a8f4bf9-36f7-409a-a509-b87dbebe361d.png" alt="secret logs.png" data-base62-sha1="1ER0Ykb3RbDnMiPrWE9voy7pKqr" width="690" height="431" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/0ba07351248dab72f7ebb288de0f0aa7c831cf07_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">secret logs.png</span><span class="informations">829×519 33.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Can you run <strong>windows-docker-image.yaml</strong> again, check whether it could succeed? If it also fail, please make sure you still have <strong>DOCKERHUB_TOKEN</strong> in secrets .</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/cd2dfd2431ca491782320f02228638614c8920e3.png" data-download-href="/uploads/short-url/th6pxYguTuURB21fnyvPKuHO69Z.png?dl=1" title="secrets.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181091964-faaaf1db-1a82-469e-9698-36b2ae0fc25b.png" alt="secrets.png" data-base62-sha1="th6pxYguTuURB21fnyvPKuHO69Z" width="690" height="338" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/cd2dfd2431ca491782320f02228638614c8920e3_2_690x338.png, https://aws1.discourse-cdn.com/github/original/2X/c/cd2dfd2431ca491782320f02228638614c8920e3.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/c/cd2dfd2431ca491782320f02228638614c8920e3.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/cd2dfd2431ca491782320f02228638614c8920e3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">secrets.png</span><span class="informations">1027×504 18.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by bfloch** [comment]
<p>We’re not sure what changed on your end but the jobs now magically started to work.</p>
<p>Thanks!</p>

**Comment by bfloch** [comment]
<p>It happened again after a PR here:</p>
<p><a href="https://github.com/nerdvegas/rez/commit/064cd43a69025bae106ce714ebb95107b45a5830/checks?check_suite_id=302811678" rel="nofollow noopener">https://github.com/nerdvegas/rez/commit/064cd43a69025bae106ce714ebb95107b45a5830/checks?check_suite_id=302811678</a></p>
<p>We did not touch the secret so we are pretty sure that it’s a bug in githubs infrastructure.</p>
